### PR TITLE
Install local cache of XML schemas on 'dashing'

### PR DESCRIPTION
**Pull request by pbaughman**
This is the 'dashing' version of [this PR](https://github.com/ros2/sros2/pull/158)

### CODE REVIEW

**Comment by pbaughman** [comment]
@kyrofa sros2 already has a [runtime dependency on rclpy](https://github.com/ros2/sros2/blob/dashing/sros2/package.xml#L13), which itself has a [runtime dependency on ament_index_python](https://github.com/ros2/rclpy/blob/dashing/rclpy/package.xml#L26), so anybody running sros2 should already have ament_index_python available at runtime.

**Comment by kyrofa** [comment]
> anybody running sros2 should already have ament_index_python available at runtime.

Ah, you covered your bases. I was about to start in on how `apt-get upgrade` won't install this update with the new dependency, but I don't think it'll complain if the dependency is already installed.

File: sros2/setup.py [hunk 1]
```diff
@@ -18,10 +19,18 @@ def package_files(directory):
...
+
 setup(
-    name='sros2',
+    name=package_name,
     version='0.6.2',
     packages=find_packages(exclude=['test']),
+    data_files=[
+        ('share/' + package_name, ['package.xml']),
+        ('share/ament_index/resource_index/packages',
+            ['resource/' + package_name]),
```
Comment (kyrofa) [inline sros2/setup.py]: These lines are fixing an unrelated issue-- mind extracting them into their own PR? Thanks for the fix, though!

Comment (pbaughman) [reply sros2/setup.py]: @kyrofa Yeah, I can open a PR for 
9b20326 and then rebase 045699d once it's merged

Comment (pbaughman) [reply sros2/setup.py]: @kyrofa Done https://github.com/ros2/sros2/pull/162

Comment (pbaughman) [reply sros2/setup.py]: @kyrofa Ok, this one is rebased now so it just contains one commit relevant to the XML catalog

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8241/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4209/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6723/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8132/)

**Pull request merged**